base/Base/Base_Base_implementation.F90

```diff
@@ -2584,158 +2760,19 @@ module subroutine MAPL_GetHorzIJIndex(npts,II,JJ,lon,lat,lonR8,latR8,Grid, rc)
-       target_lats = latR8
-    end if
-
-    _ASSERT(localSearch,"Global Search for IJ not implemented")
-
-!AOO change tusing GridType atribute    if (im_world*6==jm_world) then
-    call ESMF_AttributeGet(grid, name='GridType', value=grid_type, _RC)
-    if(trim(grid_type) == "Cubed-Sphere") then
-       call ESMF_GridGetCoord(grid,coordDim=1, localDe=0, &
-            staggerloc=ESMF_STAGGERLOC_CENTER, fArrayPtr = lons, _RC)
-       call ESMF_GridGetCoord(grid,coordDim=2, localDe=0, &
-            staggerloc=ESMF_STAGGERLOC_CENTER, fArrayPtr = lats, _RC)
-       call ESMF_GridGet(grid,coordSys=coordSys,_RC)
-       allocate(corner_lons(im+1,jm+1))
-       allocate(corner_lats(im+1,jm+1))
-       allocate(center_lons(im,jm),center_lats(im,jm))
-
-       if (coordSys==ESMF_COORDSYS_SPH_DEG) then
-          center_lons=lons*MAPL_DEGREES_TO_RADIANS_R8
-          center_lats=lats*MAPL_DEGREES_TO_RADIANS_R8
-       else if (coordSys==ESMF_COORDSYS_SPH_RAD) then
-          center_lons=lons
-          center_lats=lats
-       else if (coordSys==ESMF_COORDSYS_CART) then
-          _FAIL('Unsupported coordinate system:  ESMF_COORDSYS_CART')
-       end if
-       call MAPL_GridGetCorners(Grid,corner_lons,corner_lats,_RC)
-       ii=-1
-       jj=-1
-       call get_points_in_spherical_domain(center_lons,center_lats,corner_lons,corner_lats,target_lons,target_lats,ii,jj,_RC)
-       deallocate(corner_lons,corner_lats, center_lons,center_lats)
-    else
-       if (localSearch) then
-          call ESMF_GridGetCoord(grid,coordDim=1, localDe=0, &
-               staggerloc=ESMF_STAGGERLOC_CORNER, fArrayPtr = lons, _RC)
-          call ESMF_GridGetCoord(grid,coordDim=2, localDe=0, &
-               staggerloc=ESMF_STAGGERLOC_CORNER, fArrayPtr = lats, _RC)
-       else
-          _FAIL('if not isCubed, localSearch must be .true.')
-       end if
-       allocate(elons(im+1),_STAT)
-       allocate(elats(jm+1),_STAT)
-       call ESMF_GridGet(grid,coordSys=coordSys,_RC)
-       elons = lons(:,1)
-       elats = lats(1,:)
-       if (coordSys==ESMF_COORDSYS_SPH_DEG) then
-          elons=elons*MAPL_DEGREES_TO_RADIANS_R8
-          elats=elats*MAPL_DEGREES_TO_RADIANS_R8
-       else if (coordSys==ESMF_COORDSYS_CART) then
-          _FAIL('Unsupported coordinate system:  ESMF_COORDSYS_CART')
-       end if
-       ! lat-lon grid goes from -180 to 180 shift if we must
-       ! BMA this -180 to 180 might change at some point
-       do i=1,npts
-          lonloc = target_lons(i)
-          latloc = target_lats(i)
-          if (lonloc > MAPL_PI) lonloc = lonloc - 2.0*MAPL_PI
-          IIloc = ijsearch(elons,im+1,lonloc,.false.)
-          JJloc = ijsearch(elats,jm+1,latloc,.false.)
-          II(i) = IIloc
-          JJ(i) = JJloc
-       end do
-       deallocate(elons,elats)
-    end if
-
-    _RETURN(ESMF_SUCCESS)
+    integer                    :: status, i1, i2, j1, j2
 
-  contains
+    if (npts == 0 ) then
+      _RETURN(_SUCCESS)
+    endif
```

Comment:
You cannot return at here for the PET with zero points,  because  the `call ESMF_GridGet(grid,coordSys=coordSys,_RC)  L2826` and 
`good_grid = grid_is_ok(grid)  L2833` will crash afterwards in subroutine MAPL_GetGlobalHorzIJIndex.